controller/index.js

```diff
@@ -145,47 +143,22 @@ var Controller = function (agol, BaseController) {
     var options = req.query
     var id = params.id
     var item = params.item
-    var key = params.key
 
-    agol.find(id, function (err, data) {
-      if (err) {
-        return callback(err, null)
-      }
+    // Get the item
+    if (!parseInt(options.layer, 0)) {
+      options.layer = 0
+    }
 
-      // Get the item
-      if (!parseInt(options.layer, 0)) {
-        options.layer = 0
+    agol.getItemData(req.portal, id, item, req.key, options, function (error, itemJson) {
+      if (itemJson && itemJson.koop_status === 'processing' && typeof req.params.silent === 'undefined') {
```

Comment:
it helps to name conditionals for readability & future refactoring

```js
if (itemJson && itemJson.koop_status === 'processing' && typeof req.params.silent === 'undefined') {}
```

can be better understood as

```js
var itemExists = typeof itemJson !== 'undefined'
var isProcessing = itemExists && itemJson.koop_status === 'processing'
var silent = typeof req.params.silent !== 'undefined'

if (isProcessing && !silent) {
  return controller._returnProcessing(req, res, itemJson, callback)
}
```